src/registries/npm-registry.js

```diff
@@ -98,7 +98,7 @@ export default class NpmRegistry extends Registry {
   async getPossibleConfigLocations(filename: string): Promise<Array<[boolean, string, string]>> {
     const possibles = [
       [false, path.join(this.cwd, filename)],
-      [true, this.config.userconfig || path.join(userHome, filename)],
+      [true, this.config.userconfig || path.join(process.env.YARN_HOME_FOLDER || userHome, filename)],
```

Comment:
Before coming up with this fix, I tried several others. Most notably I tried to pass on the config to override `this.config.userconfig` used on the same line. However, `this.config` is initialized as empty object in the `constructor` of the Registries. Passing on an `initialConfig` would work, but in [`cli/commands/global.js`](https://github.com/yarnpkg/yarn/blob/6656ee3717330fb93c616c0ce8d0d9c154585774/src/cli/commands/global.js#L39) this config is discarded and rewritten. Therefore that solution would require passing on config objects all over the place in order to properly inject the location.